azurecompute/src/test/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapterLiveTest.java

```diff
@@ -0,0 +1,139 @@
+public class AzureComputeServiceAdapterLiveTest extends BaseAzureComputeApiLiveTest {
+
+   private AzureComputeServiceAdapter adapter;
+   private TemplateBuilder templateBuilder;
+   private Factory sshFactory;
+   private NodeAndInitialCredentials<Deployment> deployment;
+
+   @Override
+   protected AzureComputeApi create(Properties props, Iterable<Module> modules) {
+      Injector injector = newBuilder().modules(modules).overrides(props).buildInjector();
+      adapter = injector.getInstance(AzureComputeServiceAdapter.class);
+      templateBuilder = injector.getInstance(TemplateBuilder.class);
+      sshFactory = injector.getInstance(SshClient.Factory.class);
+      return injector.getInstance(AzureComputeApi.class);
+   }
+
+   @Test
+   public void testListLocations() {
+      Iterable<Location> locations = adapter.listLocations();
+      assertFalse(Iterables.isEmpty(locations), "locations must not be empty");
+   }
+
+   @Test
+   public void testListImages() {
+      assertFalse(Iterables.isEmpty(adapter.listImages()), "images must not be empty");
+   }
+
+   private static final PrioritizeCredentialsFromTemplate prioritizeCredentialsFromTemplate = new PrioritizeCredentialsFromTemplate(
+         new DefaultCredentialsFromImageOrOverridingCredentials());
+
+   @Test
+   public void testCreateNodeWithGroupEncodedIntoNameThenStoreCredentials() {
+      String group = "foo";
+      String name = "node" + new Random().nextInt();
+
+      Template template = templateBuilder.build();
+
+      deployment = adapter.createNodeWithGroupEncodedIntoName(group, name, template);
+      assertEquals(deployment.getNode().name(), name);
+      assertEquals(deployment.getNodeId(), deployment.getNode().name());
+      assert InetAddresses.isInetAddress(deployment.getNode().virtualIPs().get(0).address()) : deployment;
+      doConnectViaSsh(deployment.getNode(), prioritizeCredentialsFromTemplate.apply(template, deployment.getCredentials()));
+   }
+
+   protected void doConnectViaSsh(Deployment deployment, LoginCredentials creds) {
+      SshClient ssh = sshFactory.create(HostAndPort.fromParts(deployment.virtualIPs().get(0).address(), 22), creds);
+      try {
+         ssh.connect();
+         ExecResponse hello = ssh.exec("echo hello");
+         assertEquals(hello.getOutput().trim(), "hello");
+         System.err.println(ssh.exec("df -k").getOutput());
+         System.err.println(ssh.exec("mount").getOutput());
+         System.err.println(ssh.exec("uname -a").getOutput());
+      } finally {
+         if (ssh != null)
+            ssh.disconnect();
+      }
+   }
+
+   @Test
+   public void testListHardwareProfiles() {
+      Iterable<RoleSize> roleSizes = adapter.listHardwareProfiles();
+      assertFalse(Iterables.isEmpty(roleSizes));
+
+      for (RoleSize roleSize : roleSizes) {
+         assertNotNull(roleSize);
+      }
+   }
+
+   @AfterGroups(groups = "live")
```

Comment:
Also configure it with an `alwayrRun = true` to make sure we don't leak resources when the tests fail.